src/common/module/cloudsc_aux_type_mod.F90

```diff
@@ -0,0 +1,259 @@
+  USE FIELD_FACTORY_MODULE, ONLY: FIELD_NEW, FIELD_DELETE
+
+  IMPLICIT NONE
+
+  TYPE CLOUDSC_AUX_TYPE
+  
+    INTEGER(KIND=JPIM) :: NLEV
+    LOGICAL :: PACKED
+    ! 2D Fields
+    REAL(KIND=JPRB), POINTER, CONTIGUOUS :: PLSM(:)       ! Land fraction (0-1)
+    LOGICAL,         POINTER, CONTIGUOUS :: LDCUM(:)      ! Convection active
+    INTEGER(KIND=JPIM), POINTER, CONTIGUOUS :: KTYPE(:)   ! Convection type 0,1,2
+    ! 3D Fields   
+    REAL(KIND=JPRB), POINTER, CONTIGUOUS :: PLCRIT_AER(:,:)
+    REAL(KIND=JPRB), POINTER, CONTIGUOUS :: PICRIT_AER(:,:)
+    REAL(KIND=JPRB), POINTER, CONTIGUOUS :: PRE_ICE(:,:)
+    REAL(KIND=JPRB), POINTER, CONTIGUOUS :: PCCN(:,:)     ! liquid cloud condensation nuclei
+    REAL(KIND=JPRB), POINTER, CONTIGUOUS :: PNICE(:,:)    ! ice number concentration (cf. CCN)
+    REAL(KIND=JPRB), POINTER, CONTIGUOUS :: PT(:,:)       ! T at start of callpar
+    REAL(KIND=JPRB), POINTER, CONTIGUOUS :: PQ(:,:)       ! Q at start of callpar
+    REAL(KIND=JPRB), POINTER, CONTIGUOUS :: PVFA(:,:)     ! CC from VDF scheme
+    REAL(KIND=JPRB), POINTER, CONTIGUOUS :: PVFL(:,:)     ! Liq from VDF scheme
+    REAL(KIND=JPRB), POINTER, CONTIGUOUS :: PVFI(:,:)     ! Ice from VDF scheme
+    REAL(KIND=JPRB), POINTER, CONTIGUOUS :: PDYNA(:,:)    ! CC from Dynamics
+    REAL(KIND=JPRB), POINTER, CONTIGUOUS :: PDYNL(:,:)    ! Liq from Dynamics
+    REAL(KIND=JPRB), POINTER, CONTIGUOUS :: PDYNI(:,:)    ! Liq from Dynamics
+    REAL(KIND=JPRB), POINTER, CONTIGUOUS :: PHRSW(:,:)    ! Short-wave heating rate
+    REAL(KIND=JPRB), POINTER, CONTIGUOUS :: PHRLW(:,:)    ! Long-wave heating rate
+    REAL(KIND=JPRB), POINTER, CONTIGUOUS :: PVERVEL(:,:)  ! Vertical velocity
+    REAL(KIND=JPRB), POINTER, CONTIGUOUS :: PAP(:,:)      ! Pressure on full levels
+    REAL(KIND=JPRB), POINTER, CONTIGUOUS :: PAPH(:,:)     ! Pressure on half levels
+    REAL(KIND=JPRB), POINTER, CONTIGUOUS :: PLU(:,:)      ! Conv. condensate
+    REAL(KIND=JPRB), POINTER, CONTIGUOUS :: PLUDE(:,:)    ! Conv. detrained water
+    REAL(KIND=JPRB), POINTER, CONTIGUOUS :: PSNDE(:,:)    ! Conv. detrained snow
+    REAL(KIND=JPRB), POINTER, CONTIGUOUS :: PMFU(:,:)     ! Conv. mass flux up
+    REAL(KIND=JPRB), POINTER, CONTIGUOUS :: PMFD(:,:)     ! Conv. mass flux down
+    REAL(KIND=JPRB), POINTER, CONTIGUOUS :: PA(:,:)       ! Original Cloud fraction (t)
+    REAL(KIND=JPRB), POINTER, CONTIGUOUS :: PEXTRA(:,:,:) ! extra fields
+    REAL(KIND=JPRB), POINTER, CONTIGUOUS :: PCLV(:,:,:)
+    REAL(KIND=JPRB), POINTER, CONTIGUOUS :: PSUPSAT(:,:)
+    ! Output fields used for validation
+    REAL(KIND=JPRB), POINTER, CONTIGUOUS :: PCOVPTOT(:,:) ! Precip fraction
+    REAL(KIND=JPRB), POINTER, CONTIGUOUS :: PRAINFRAC_TOPRFZ(:)
+    
+    ! field gang for 3D fields
+    CLASS(FIELD_4RB), POINTER :: DATA_RDONLY
+    ! Acces pointers for 3D fields in gang
+    TYPE(FIELD_3RB_PTR), ALLOCATABLE :: FIELDS_RDONLY(:)
```

Comment:
This should never be accessed by external code, so it should be marked `PRIVATE`.

Reply:
Currently I am calling the sync_device_rdonly() of the field gang from the driver, and this would have to be changed if I made this private. I can add a method to the aux, flux, and tendency objects to expose the sync method of the field gang in the driver if it is preferable. What do you think? @awnawab

Reply:
Sorry I realise now my comment was quite ambiguous. Whilst the FieldGang itself should remain publicly accessible, `FIELDS_RDONLY` which is used only to store pointers to the gang's children, should be made private. There shouldn't be a need to call this from external code.